hudi-common/src/main/java/org/apache/hudi/BaseHoodieTableFileIndex.java

```diff
@@ -360,13 +363,22 @@ protected HoodieTimeline getActiveTimeline() {
   }
 
   private Object[] parsePartitionColumnValues(String[] partitionColumns, String partitionPath) {
-    Object[] partitionColumnValues = doParsePartitionColumnValues(partitionColumns, partitionPath);
-    if (shouldListLazily && partitionColumnValues.length != partitionColumns.length) {
-      throw new HoodieException("Failed to parse partition column values from the partition-path:"
-          + " likely non-encoded slashes being used in partition column's values. You can try to"
-          + " work this around by switching listing mode to eager");
+    HoodieTableConfig tableConfig = metaClient.getTableConfig();
+    Object[] partitionColumnValues;
+    if (null != tableConfig.getKeyGeneratorClassName()
+        && tableConfig.getKeyGeneratorClassName().equals(KeyGeneratorType.TIMESTAMP.getClassName())
+        && tableConfig.propsMap().get(TimestampKeyGeneratorConfig.TIMESTAMP_TYPE_FIELD.key()).matches("SCALAR|UNIX_TIMESTAMP|EPOCHMILLISECONDS")) {
+      // For TIMESTAMP key generator when TYPE is SCALAR, UNIX_TIMESTAMP or EPOCHMILLISECONDS,
+      // we couldn't reconstruct initial partition column values from partition paths due to lost data after formatting in most cases
+      partitionColumnValues = new Object[partitionColumns.length];
```

Comment:
Partition column values are empty.  Can this cause the partition pruning to return wrong or empty results from `SparkHoodieTableFileIndex::listMatchingPartitionPaths`?

Reply:
I will check it.

Reply:
@yihua, you are right, it could lead to an empty result, but the problem is with `prunedPartitionsAndFilteredFileSlices` in `HoodieFileIndex::listFiles`. Null partition column values lead to `PartitionDirectory`s with null values due to used
```
InternalRow.fromSeq(partitionOpt.get.values)
```
I opened a separate related task: HUDI-7952, and will search for proper fixing.

Reply:
@geserdugarov @yihua I am reverting this patch to unblock 1.0.0-beta2 release testing. I will reopen HUDI-7709. Let's discuss there.

Reply:
@geserdugarov , do you think we have a proper fix plan now, it looks like this one is suspended.

Reply:
> Execution of query like:
SELECT ... WHERE ts BETWEEN 1078016000 and 1718953003 ...
will pass an empty result even with passed partition paths as UTF8Strings, because we need to deal with initial UNIX_TIMESTAMP values for this filtering.

> Need to think about how to make it work.

It looks like there is no good way to fix this because of the datetime string format, either we handle the partition paths from the query predicates with the same key generator and use that for pruning or we store the initial non-formatted partition values.

Reply:
@danny0405 , as I remember, the problem that there was no way to force not to do partition pruning when it is not suitable.
Rechecked it on current master, 41816e30041b82da7505c8c18288cf4f5df4e00a, using PySpark:
HUDI-7952/forced-partition-pruning-workaround.py

Reply:
As I mentioned in comment:
https://github.com/apache/hudi/blob/41816e30041b82da7505c8c18288cf4f5df4e00a/hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestSparkSqlWithTimestampKeyGenerator.scala#L88-L89
we need to support `shouldExtractPartitionValuesFromPartitionPath` as `false` in this case, and process correspondingly. We shouldn't do any partition pruning in this case, and retrieving all data for `ts` column from data files.

Reply:
Here Jon wrote, that pruning is still not working properly:
https://github.com/apache/hudi/pull/11770/files#diff-96c57b7af6bc51184f738fcad26b2b966e5f899a3a9eaf1d99dfd5d09dc301fcR92

Reply:
> We shouldn't do any partition pruning in this case, and retrieving all data for ts column from data files.

This is right because the un-formatted partition value are only stored in the files.